src/messageHandler.ts

```diff
@@ -37,7 +37,9 @@ type ClientCallbacks = {
   onInvestmentSellRequested?: (event: InvestmentSellRequestedEvent) => Promise<InvestmentSellRequestedMessage>;
   onShare?: (event: ShareEvent) => Promise<any>;
   onExitRequested: (event: ExitRequestedEvent) => Promise<any>;
-  onTransferConsentRequested?: (event: TransferConsentRequestedEvent) => Promise<TransferConsentRequestedMessage>;
+  onTransferConsentRequested?: (
+    event: TransferConsentRequestedEvent,
+  ) => Promise<TransferConsentRequestedMessage | undefined>;
```

Comment:
Allow returning `undefined` - in which case we do not `postTransferConsentRequestSucceeded` immediately (instead we will let the partner navigate to the success URL when they are ready)